test/util/bspNet/waits.ts

```diff
@@ -323,30 +310,47 @@ export const waitForBspToCatchUpToChainTip = async (
  * @throws Will throw an error if the expected extrinsic or event is not found.
  */
 export const waitForMspResponseWithoutSealing = async (api: ApiPromise, checkQuantity?: number) => {
-  const iterations = 41;
-  const delay = 50;
+  await waitForTxInPool(api, {
+    module: "fileSystem",
+    method: "mspRespondStorageRequestsMultipleBuckets",
+    checkQuantity,
+    iterations: 41,
+    delay: 50
+  });
+};
+
+/**
+ * Options for the `waitFor` function.
+ * @param lambda - The condition to wait for.
+ * @param iterations - The number of iterations to wait for the condition to be true.
+ * @param delay - The delay between iterations.
+ */
+export interface WaitForOptions {
+  lambda: () => Promise<boolean>;
+  iterations?: number;
+  delay?: number;
+}
+
+/**
+ * Waits for an arbitrary condition to be true. It keeps polling the condition until it is true or
+ * a timeout is reached.
+ */
+export const waitFor = async (options: WaitForOptions) => {
```

Comment:
REALLY LIKE THIS. Much needed.

Reply:
Thanks. Eventually I did a drive-by refactor in the `waits.ts` since it was bugging me too much.